src/briefcase/integrations/subprocess.py

```diff
@@ -416,17 +440,15 @@ def _log_environment(self, overrides):
         :param overrides: The explicit environment passed to the subprocess call;
             can be `None` if there are no explicit environment changes.
         """
-        if self.command.logger.verbosity >= self.command.logger.DEEP_DEBUG:
-            full_env = self.full_env(overrides)
-            self.command.logger.deep_debug("Full Environment:")
-            for env_var, value in full_env.items():
-                self.command.logger.deep_debug(f"    {env_var}={value}")
-
-        elif self.command.logger.verbosity >= self.command.logger.DEBUG:
-            if overrides:
-                self.command.logger.debug("Environment:")
-                for env_var, value in overrides.items():
-                    self.command.logger.debug(f"    {env_var}={value}")
+        full_env = self.full_env(overrides, sanitize=True)
+        self.command.logger.deep_debug("Full Environment:")
+        for env_var, value in full_env.items():
+            self.command.logger.deep_debug(f"    {env_var}={value}")
+
+        if overrides:
+            self.command.logger.debug("Environment Overrides:")
+            for env_var, value in overrides.items():
+                self.command.logger.debug(f"    {env_var}={value}")
```

Comment:
I think we should do a rethink of "full" environment handling.

Previously, the motivation for always including the full environment was that users are prone to only giving us a part of the full log when they report problems, so it's in our interests as maintainers to ensure that we include the full environment every time to minimise the chance that someone will redact the useful information when reporting. 

However, this PR gives us a clean way to direct users to give us the full log with no redactions, so that problem no longer exists. On that basis, I think we only need an initial dump of the "ambient" environment; we can then limit the "in stream" environment logs to the overrides. That makes the log a lot easier to read, as the ambient environment becomes part of the preamble, and the log itself only contains "new" information.

I'd even argue `-vv` may not be required any more, as it primarily existed to make sure we get that full environment.

Reply:
💯 

Completely agree. I moved full environment logging for subprocess commands to creating the log file. I also removed `deep_debug` logging functionality and updated the help text for `-v`.